jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -2160,6 +2160,9 @@ static int indexOf(byte[] value, byte coder, int count, String str, int fromInde
 		if (s2Length == 0) {
 			// At this point we know fromIndex < s1Length so there is a hit at fromIndex
```

Comment:
While you're here, can you fix this comment, please: `<` should be `<=`.

Reply:
Isn't it correct here? On Line 2154 we have `else if (fromIndex >= s1Length)` check which returns early, so when we get here the reverse must be true. `fromIndex < s1Length`.

Reply:
Not strictly correct, no. If `fromIndex < 0` at the call site, we take the path that sets `fromIndex` to zero without comparing it with `s1Length`. If `s1Length` is zero, then `fromIndex == s1Length`.